Loop/Managers/LoopDataManager.swift

```diff
@@ -847,23 +891,85 @@ extension LoopDataManager {
             throw LoopError.missingDataError(.glucose)
         }
 
+        let pumpStatusDate = doseStore.lastAddedPumpData
```

Comment:
[nit] Why separate constance for `doseStore.lastAddedPumpData`, but not `glucose.startDate` (line 898)?

Reply:
Might want to move line 916 up here. Specifically, `let lastGlucoseDate = glucose.startDate`